tests/models/test_hparams.py

```diff
@@ -642,3 +649,30 @@ def test_model_with_fsspec_as_parameter(tmpdir):
     )
     trainer.fit(model)
     trainer.test()
+
+
+@pytest.mark.skipif(not HYDRA_AVAILABLE, reason="Hydra is not available")
+def test_model_save_hyper_parameters_interpolation_with_hydra(tmpdir):
+
+    initialize(config_path="conf")
+
+    cfg = compose(config_name="config")
+
+    class TestModel(BoringModel):
+
+        def __init__(self, cfg):
+            self.save_hyperparameters()
+            assert isinstance(self.hparams.cfg, DictConfig)
+            assert self.hparams.cfg.log == "Something"
+            super().__init__()
+
+    model = TestModel(cfg)
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        limit_train_batches=2,
+        limit_val_batches=2,
+        limit_test_batches=2,
+        max_epochs=1,
+    )
+    trainer.fit(model)
+    trainer.test()
```

Comment:
Do we need to run `fit` and `test`? we are not asserting anything for that

```suggestion
```

Reply:
Yeah good point, @tchaton you might be able to just get away with ```Trainer.save_checkpoint(...)``` and re-loading the model for comparison.

Atm this implicitly saves the model and ensures it doesn't crash

Reply:
Removed ! Don't think there is a need. We already have test for DictConfig. I just needed to make sure the interpolation was properly performed.

Reply:
Done !